app/jmeter/jsd.jmx

```diff
@@ -117,7 +117,52 @@
         <stringProp name="ThreadGroup.delay"></stringProp>
         <boolProp name="ThreadGroup.same_user_on_next_iteration">true</boolProp>
       </SetupThreadGroup>
-      <hashTree/>
+      <hashTree>
+        <TestAction guiclass="TestActionGui" testclass="TestAction" testname="prepare" enabled="true">
+          <intProp name="ActionProcessor.action">1</intProp>
+          <intProp name="ActionProcessor.target">0</intProp>
+          <stringProp name="ActionProcessor.duration">0</stringProp>
+        </TestAction>
+        <hashTree>
+          <BeanShellPreProcessor guiclass="TestBeanGUI" testclass="BeanShellPreProcessor" testname="prepare customer requests" enabled="true">
+            <boolProp name="resetInterpreter">false</boolProp>
+            <stringProp name="parameters"></stringProp>
+            <stringProp name="filename"></stringProp>
+            <stringProp name="script">import org.apache.commons.collections.iterators.ArrayIterator;
+import org.apache.commons.io.FileUtils;
+
+
+    Map portalRequestsPairByCustomerUsername = new HashMap();
+    // TODO Rework hard-codded path
+    List rowList = FileUtils.readLines(new File(&quot;datasets/jsd/customers.csv&quot;));
+    for (Object o: rowList) {
+        String line = (String) o;
+        String[] row = line.split(&quot;,&quot;);
+        int requestsStartOrderNum = 2;
+        if (row.length &lt;= requestsStartOrderNum) {
+            continue;
+        }
+
+        String[] portalRequestRawPairs = Arrays.copyOfRange(row, requestsStartOrderNum, row.length);
+
+        ArrayIterator iterator = new ArrayIterator(portalRequestRawPairs);
+        List pairs = new ArrayList();
+        while (iterator.hasNext()) {
+            List pair = new ArrayList();
+            pair.add(iterator.next());
+            pair.add(iterator.next());
```

Comment:
Why we use pair.add(iterator.next()); twice?

Reply:
because of two items:
1. portal id
2. request key